DataFormats/Detectors/ITSMFT/MFT/include/DataFormatsMFT/TrackParamMFT.h

```diff
@@ -0,0 +1,200 @@
+  void setRemovable(Bool_t removable) { mRemovable = removable; }
+
+  /// return the chi2 of the track when the associated cluster was attached
+  Double_t getTrackChi2() const { return mTrackChi2; }
+  /// set the chi2 of the track when the associated cluster was attached
+  void setTrackChi2(Double_t chi2) { mTrackChi2 = chi2; }
+  /// return the local chi2 of the associated cluster with respect to the track
+  Double_t getLocalChi2() const { return mLocalChi2; }
+  /// set the local chi2 of the associated cluster with respect to the track
+  void setLocalChi2(Double_t chi2) { mLocalChi2 = chi2; }
+
+  Bool_t isCompatibleTrackParamMFT(const TrackParamMFT& TrackParamMFT, Double_t sigma2Cut, Double_t& normChi2) const;
+
+  void print() const;
+
+  void clear();
+
+ private:
+  Double_t mZ = 0.; ///< Z coordinate (cm)
+
+  /// Track parameters ordered as follow:      <pre>
+  /// X       = X coordinate   (cm)
+  /// Y       = Y coordinate   (cm)
+  /// PHI     = azimutal angle
+  /// TANL    = tangent of \lambda (dip angle)
+  /// INVQPT    = Inverse transverse momentum (GeV/c ** -1) times charge (assumed forward motion)  </pre>
+  TMatrixD mParameters{5, 1}; ///< \brief Track parameters
+
+  /// Covariance matrix of track parameters, ordered as follows:    <pre>
+  ///  <X,X>         <Y,X>           <PHI,X>       <TANL,X>        <INVQPT,X>
+  ///  <X,Y>         <Y,Y>           <PHI,Y>       <TANL,Y>        <INVQPT,Y>
+  /// <X,PHI>       <Y,PHI>         <PHI,PHI>     <TANL,PHI>      <INVQPT,PHI>
+  /// <X,TANL>      <Y,TANL>       <PHI,TANL>     <TANL,TANL>     <INVQPT,TANL>
+  /// <X,INVQPT>   <Y,INVQPT>     <PHI,INVQPT>   <TANL,INVQPT>   <INVQPT,INVQPT>  </pre>
+  mutable std::unique_ptr<TMatrixD> mCovariances{}; ///< \brief Covariance matrix of track parameters
+
+  /// Jacobian used to extrapolate the track parameters and covariances to the actual z position
+  mutable std::unique_ptr<TMatrixD> mPropagator{};
+  /// Track parameters extrapolated to the actual z position (not filtered by Kalman)
+  mutable std::unique_ptr<TMatrixD> mExtrapParameters{};
+  /// Covariance matrix extrapolated to the actual z position (not filtered by Kalman)
+  mutable std::unique_ptr<TMatrixD> mExtrapCovariances{};
+
+  mutable std::unique_ptr<TMatrixD> mSmoothParameters{};  ///< Track parameters obtained using smoother
+  mutable std::unique_ptr<TMatrixD> mSmoothCovariances{}; ///< Covariance matrix obtained using smoother
+
+  const o2::itsmft::Cluster* mClusterPtr = nullptr; ///< Pointer to the associated cluster if any
+
+  Bool_t mRemovable = false; ///< kTRUE if the associated cluster can be removed from the track it belongs to
+
+  Double_t mTrackChi2 = 0.; ///< Chi2 of the track when the associated cluster was attached
+  Double_t mLocalChi2 = 0.; ///< Local chi2 of the associated cluster with respect to the track
+};
+
+class TrackMFTParam : public TrackParamMFT
```

Comment:
deriving TrackMFTParam from TrackParamMFT is somewhat confusing, perhaps more appropriate names can be used? But I don't see  where the TrackMFTParam is used at all. Is this class needed?